src/Core/Models/Business/SecretsManagerSubscriptionUpdate.cs

```diff
@@ -53,4 +56,39 @@ public class SecretsManagerSubscriptionUpdate
     public bool SmServiceAccountsChanged => SmServiceAccountsAdjustment != 0;
     public bool MaxAutoscaleSmSeatsChanged { get; set; }
     public bool MaxAutoscaleSmServiceAccountsChanged { get; set; }
+
+    public SecretsManagerSubscriptionUpdate(
+        Organization organization,
+        int seatAdjustment, int? maxAutoscaleSeats,
+        int serviceAccountAdjustment, int? maxAutoscaleServiceAccounts)
+    {
+        if (organization == null)
+        {
+            throw new NotFoundException("Organization is not found.");
+        }
+
+        var secretsManagerPlan = Utilities.StaticStore.GetSecretsManagerPlan(organization.PlanType);
+        if (secretsManagerPlan == null)
+        {
+            throw new NotFoundException("Invalid Secrets Manager plan.");
+        }
+
+        var newTotalSeats = organization.SmSeats.GetValueOrDefault() + seatAdjustment;
+        var newTotalServiceAccounts = organization.SmServiceAccounts.GetValueOrDefault() + serviceAccountAdjustment;
+
+        Organization = organization;
+
+        SmSeats = newTotalSeats;
+        SmSeatsAdjustment = seatAdjustment;
+        SmSeatsExcludingBase = newTotalSeats - secretsManagerPlan.BaseSeats;
+        MaxAutoscaleSmSeats = maxAutoscaleSeats;
+
+        SmServiceAccounts = newTotalServiceAccounts;
+        SmServiceAccountsAdjustment = serviceAccountAdjustment;
+        SmServiceAccountsExcludingBase = newTotalServiceAccounts - secretsManagerPlan.BaseServiceAccount.GetValueOrDefault();
+        MaxAutoscaleSmServiceAccounts = maxAutoscaleServiceAccounts;
+
+        MaxAutoscaleSmSeatsChanged = maxAutoscaleSeats.HasValue && maxAutoscaleSeats != organization.MaxAutoscaleSmSeats.GetValueOrDefault();
+        MaxAutoscaleSmServiceAccountsChanged = maxAutoscaleServiceAccounts.HasValue && maxAutoscaleServiceAccounts != organization.MaxAutoscaleSmServiceAccounts.GetValueOrDefault();
```

Comment:
You need to update this with the changes from https://github.com/bitwarden/server/pull/3085

Reply:
What is missing?